### PR TITLE
Add zig httpz server

### PR DESCRIPTION
**Pull request by Kayden1412**

### CODE REVIEW

**Review by waghanza — CHANGES_REQUESTED** [review]
Thanks for your contribution :tada:

Could you specify GitHub location, and any website for httpz?

**Comment by waghanza** [comment]
@karlseguin We are going to introduce httpz here. Is this something you are OK with.

We can ping you (not so often) to make sure the implementation is OK

**Comment by waghanza** [comment]
@Kayden1412 could you use https://github.com/ziglang/docker-zig if it is updated?

File: zig/httpz/src/main.zig [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    var router = server.router();
+
+    router.get("/", empty_handler);
+    router.post("/user", empty_handler);
+    router.get("/user/:id", user_id_handler);
+    try server.listen();
+}
+
+pub fn user_id_handler(req: *httpz.Request, res: *httpz.Response) !void {
+    res.keepalive = true;
```
Comment (karlseguin) [inline zig/httpz/src/main.zig:34]: don't need to set this

File: zig/httpz/src/main.zig [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    res.body = req.param("id") orelse "";
+}
+
+pub fn empty_handler(_: *httpz.Request, res: *httpz.Response) !void {
+    res.keepalive = true;
```
Comment (karlseguin) [inline zig/httpz/src/main.zig:39]: don't need to set this

**Comment by karlseguin** [comment]
Sure. 

Current master code is pretty far from 0.13 branch. Maybe when Zig 0.14 is released, we can do an update.

**Comment by waghanza** [comment]
I have not enough worked with zig to know which image suits best.

The idea is to have an image that you are confident with to go in production, at list  n image close to this goal :stuck_out_tounge:

**Comment by waghanza** [comment]
I'll add this PR for next round, and monitor zig updates

**Review by waghanza — APPROVED** [review]

**Comment by waghanza** [comment]
@karlseguin @Kayden1412  I have 
```
thread 26 panic: reached unreachable code
Panicked during a panic. Aborting.
error: the following build command crashed:
/app/.zig-cache/o/3c97310c79e5fa4f1bb00d7c1d5b2ebe/build /usr/bin/zig /app /app/.zig-cache /root/.cache/zig --seed 0x4dd03a9d -Z48d7c4c12cea8f62 -Doptimize=ReleaseFast
```

when trying to compile inside a container with ` zig build -Doptimize=ReleaseFast`

maybe something related to arch, we use aarch to produce results (mac m1)

**Pull request merged**